lib/matplotlib/backend_bases.py

```diff
@@ -2256,22 +2256,16 @@ def print_figure(
 
         # Remove the figure manager, if any, to avoid resizing the GUI widget.
         with cbook._setattr_cm(self, manager=None), \
-                cbook._setattr_cm(self.figure, dpi=dpi), \
-                cbook._setattr_cm(canvas, _is_saving=True):
-            origfacecolor = self.figure.get_facecolor()
-            origedgecolor = self.figure.get_edgecolor()
-
-            if facecolor is None:
-                facecolor = rcParams['savefig.facecolor']
-            if cbook._str_equal(facecolor, 'auto'):
-                facecolor = origfacecolor
-            if edgecolor is None:
-                edgecolor = rcParams['savefig.edgecolor']
-            if cbook._str_equal(edgecolor, 'auto'):
-                edgecolor = origedgecolor
-
-            self.figure.set_facecolor(facecolor)
-            self.figure.set_edgecolor(edgecolor)
+             cbook._setattr_cm(self.figure, dpi=dpi), \
+             cbook._setattr_cm(canvas, _is_saving=True), \
+             ExitStack() as stack:
```

Comment:
Perhaps we could re-arrange this to avoid the escaped newlines:
```suggestion
        with ExitStack() as stack:
             stack.enter_context(cbook._setattr_cm(self, manager=None))
             stack.enter_context(cbook._setattr_cm(self.figure, dpi=dpi))
             stack.enter_context(cbook._setattr_cm(canvas, _is_saving=True))
```
and maybe move the comment in.

Reply:
I think having syntactic context managers remains more readable; also note that in Py3.10+ we'll be able to parenthesize the multiple contextmanagers to avoid the backslashes anyways.

Reply:
OK